sway-lsp/src/server.rs

```diff
@@ -532,52 +532,58 @@ impl Backend {
                         None
                     };
 
+                // Returns true if the current path matches the path of a submodule
+                let path_is_submodule = |ident: &Ident, path: &Option<PathBuf>| -> bool {
+                    ident.span().path().map(|a| a.deref()) == path.as_ref()
+                };
+
                 {
                     let program = session.compiled_program.read();
                     match params.ast_kind.as_str() {
+                        "lexed" => {
```

Comment:
just out of curiosity, is there a reason we match on the string slice here rather than the `ast_kind` as is? That is, assuming it's an enum.

Reply:
Yeah that `ast_kind` is a `String`. Converting it to a `str` here just means there is less of `"lexed".to_string()` needed when matching for example.